network/phonebook.go

```diff
@@ -164,18 +170,17 @@ func (e *phonebookEntries) waitForConnectionTime(addr string) (addrInPhonebook,
 	// If there are max number of connections within the time window, wait
 	numElts := len(e.data[addr].recentConnectionTimes)
```

Comment:
I did not want to use numElts in the loop, and again down here because the number of elements may change in
e.popNElements(numElmtsToRemove, addr)
between the loop and its usage here. In the event it is not updated again here, it will be a bug.